src/main/java/ai/shreds/infrastructure/InfrastructureSupplierRepositoryImpl.java

```diff
@@ -0,0 +1,71 @@
+package ai.shreds.infrastructure;
+
+import ai.shreds.domain.DomainSupplierEntity;
+import ai.shreds.domain.DomainSupplierRepositoryPort;
+import ai.shreds.shared.SharedRequestParams;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import javax.persistence.EntityManager;
+import javax.persistence.TypedQuery;
+import java.util.List;
+import java.util.Optional;
+import ai.shreds.domain.SupplierNotFoundException;
+
+@Repository
+public class InfrastructureSupplierRepositoryImpl implements DomainSupplierRepositoryPort {
+
+    private static final Logger logger = LoggerFactory.getLogger(InfrastructureSupplierRepositoryImpl.class);
+
+    @Autowired
+    private EntityManager entityManager;
+
+    /**
+     * Finds all suppliers based on the provided filtering criteria.
+     * 
+     * @param params The filtering criteria for retrieving suppliers.
+     * @return A list of suppliers that match the filtering criteria.
+     */
+    @Override
+    public List<DomainSupplierEntity> findAll(SharedRequestParams params) {
+        logger.info("Finding all suppliers with params: {}", params);
+        StringBuilder queryStr = new StringBuilder("SELECT s FROM DomainSupplierEntity s WHERE 1=1");
+        if (params.getName() != null && !params.getName().isEmpty()) {
+            queryStr.append(" AND s.name LIKE :name");
+        }
+        if (params.getContact_info() != null && !params.getContact_info().isEmpty()) {
+            queryStr.append(" AND s.contact_info LIKE :contact_info");
+        }
+        if (params.getAddress() != null && !params.getAddress().isEmpty()) {
+            queryStr.append(" AND s.address LIKE :address");
+        }
```

Comment:
extract the code to a method returning 'queryStr' and call it in the findAll method